nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -1354,55 +1349,55 @@ private void validateSnippetContents(final FlowSnippetDTO flow) {
-                    .forEach(id -> identifiers.add(id));
-                snippet.getProcessGroups().stream()
-                    .map(group -> group.getId())
-                    .forEach(id -> identifiers.add(id));
-                snippet.getRemoteProcessGroups().stream()
-                    .map(remoteGroup -> remoteGroup.getId())
-                    .forEach(id -> identifiers.add(id));
-                snippet.getRemoteProcessGroups().stream()
-                    .flatMap(remoteGroup -> remoteGroup.getContents().getInputPorts().stream())
-                    .map(remoteInputPort -> remoteInputPort.getId())
-                    .forEach(id -> identifiers.add(id));
-                snippet.getRemoteProcessGroups().stream()
-                    .flatMap(remoteGroup -> remoteGroup.getContents().getOutputPorts().stream())
-                    .map(remoteOutputPort -> remoteOutputPort.getId())
-                    .forEach(id -> identifiers.add(id));
+                    // save the flow
+                    controllerFacade.save();
 
-                final ProcessGroup processGroup = processGroupDAO.getProcessGroup(groupId);
-                return revisionManager.get(identifiers,
-                    () -> {
-                        final ProcessGroupStatus groupStatus = controllerFacade.getProcessGroupStatus(groupId);
-                        return dtoFactory.createFlowDto(processGroup, groupStatus, snippet, revisionManager);
-                    });
-            });
+                    // drop the snippet
+                    snippetDAO.dropSnippet(snippetId);
+
+                    // identify all components added
+                    final Set<String> identifiers = new HashSet<>();
+                    snippet.getProcessors().stream()
+                            .map(proc -> proc.getId())
+                            .forEach(id -> identifiers.add(id));
+                    snippet.getConnections().stream()
+                            .map(conn -> conn.getId())
+                            .forEach(id -> identifiers.add(id));
+                    snippet.getInputPorts().stream()
+                            .map(port -> port.getId())
+                            .forEach(id -> identifiers.add(id));
+                    snippet.getOutputPorts().stream()
+                            .map(port -> port.getId())
+                            .forEach(id -> identifiers.add(id));
+                    snippet.getProcessGroups().stream()
+                            .map(group -> group.getId())
+                            .forEach(id -> identifiers.add(id));
+                    snippet.getRemoteProcessGroups().stream()
+                            .map(remoteGroup -> remoteGroup.getId())
+                            .forEach(id -> identifiers.add(id));
+                    snippet.getRemoteProcessGroups().stream()
+                            .flatMap(remoteGroup -> remoteGroup.getContents().getInputPorts().stream())
```

Comment:
I went ahead and addressed the issue myself. Take a look and make sure we're on the same page.

https://github.com/mcgilman/nifi/commit/6daea58b171368bede33248fd2d4a56d295c1687